packages/react-environment/src/api-providers/polkadot/polkadot-bridge-deposit.ts

```diff
@@ -0,0 +1,101 @@
+import { ChainType, computeChainIdType, InternalChainId, SubstrateChainId } from '@webb-dapp/apps/configs';
+import { DepositPayload as IDepositPayload, MixerSize } from '@webb-dapp/react-environment';
+import { WebbPolkadot } from '@webb-dapp/react-environment/api-providers/polkadot/webb-polkadot-provider';
+import { BridgeConfig } from '@webb-dapp/react-environment/types/bridge-config.interface';
+import { BridgeApi } from '@webb-dapp/react-environment/webb-context/bridge/bridge-api';
+import { WebbError, WebbErrorCodes } from '@webb-dapp/utils/webb-error';
+import { Note, NoteGenInput } from '@webb-tools/sdk-core';
+
+import { u8aToHex } from '@polkadot/util';
+
+import { BridgeDeposit } from '../../webb-context/bridge/bridge-deposit';
+
+type DepositPayload = IDepositPayload<Note, [number, string]>;
+
+export class PolkadotBridgeDeposit extends BridgeDeposit<WebbPolkadot, DepositPayload> {
+  async deposit(depositPayload: DepositPayload): Promise<void> {
+    const tx = this.inner.txBuilder.build(
+      {
+        section: 'anchorBn254',
+        method: 'deposit',
+      },
+      depositPayload.params
+    );
+    const account = await this.inner.accounts.activeOrDefault;
+    if (!account) {
+      throw WebbError.from(WebbErrorCodes.NoAccountAvailable);
+    }
+    const hash = await tx.call(account.address);
+    return;
+  }
+
+  async generateBridgeNote(
+    mixerId: number | string,
+    destination: number,
+    wrappableAssetAddress: string | undefined
+  ): Promise<DepositPayload> {
+    const currency = this.bridgeApi.currency;
+
+    if (!currency) {
+      throw new Error('api not ready');
+    }
+    const tokenSymbol = currency.view.symbol;
+    const destChainId = destination;
+    // TODO: add mappers similar to evm chain id
+    // const chainId = this.inner.api.registry.chainSS58!;
+    const chainId = SubstrateChainId.Webb;
+    const sourceChainId = computeChainIdType(ChainType.Substrate, chainId);
+    const anchorPath = String(mixerId).replace('Bridge=', '').split('@');
+    const amount = anchorPath[0];
+    const anchorIndex = anchorPath[2];
+    const anchors = await this.bridgeApi.getAnchors();
+    const anchor = anchors[Number(anchorIndex)];
+    console.log({
+      amount,
+      anchorIndex,
+      anchor,
+      anchors,
+      sourceChainId,
+      destination,
+      mixerId,
+    });
```

Comment:
Thoughts on making use of the structured logger here instead of just `console`? I tend to prefer using a structured logger in all cases so that you may specify the severity/case. More opinionated than anything.

Reply:
Yeah you are right, but I meant to remove those in the future but I guess it's a good idea to keep them as trace logs